src/ngraph/frontend/onnx_import/core/tensor.hpp

```diff
@@ -109,11 +107,36 @@ namespace ngraph
                             return {std::begin(container), std::end(container)};
                         }
 
+                        /// Returns the size if bytes of an ONNX data type.
+                        inline size_t __get_onnx_data_size(int data_type)
+                        {
+                            switch (data_type)
+                            {
+                            case onnx::TensorProto_DataType_FLOAT: return sizeof(float);
+                            case onnx::TensorProto_DataType_UINT8: return sizeof(uint8_t);
+                            case onnx::TensorProto_DataType_INT8: return sizeof(int8_t);
+                            case onnx::TensorProto_DataType_UINT16: return sizeof(uint16_t);
+                            case onnx::TensorProto_DataType_INT16: return sizeof(int16_t);
+                            case onnx::TensorProto_DataType_INT32: return sizeof(int32_t);
+                            case onnx::TensorProto_DataType_INT64: return sizeof(int64_t);
+                            case onnx::TensorProto_DataType_BOOL: return sizeof(bool);
+                            case onnx::TensorProto_DataType_FLOAT16: return 2;
```

Comment:
Can we use `sizeof(ngraph::float16)` instead of hard-coding `2`?

Reply:
Technically, we should be verifying that the actual sizes in ONNX match the sizes of the types here. For example, `float` isn't guaranteed to be 32 bits or even IEEE format.